conjure-java-core/src/main/java/com/palantir/conjure/java/services/UndertowServiceHandlerGenerator.java

```diff
@@ -519,12 +528,70 @@ private CodeBlock generateParameterCodeBlock(Stream<ArgumentDefinition> params,
                                         arg.getArgName().get(),
                                         createConstructorForTypeWithReference(arg.getType(), rawVarName,
                                                 typeDefinitions, typeMapper)
-                                )
+                                ),
+                                generateStoreParamInExchange(arg)
                         );
                     }
                 }).collect(Collectors.toList()));
     }
 
+    private Optional<String> getStoreMethodAndParamName(ArgumentDefinition arg) {
+        boolean isSafe = arg.getMarkers().stream().anyMatch(type ->
+                type.accept(
+                        new DefaultTypeVisitor<Boolean>() {
+                            @Override
+                            public Boolean visitExternal(ExternalReference value) {
+                                return value.getExternalReference().getName().equals("Safe");
```

Comment:
This should probably get the class name of this marker to verify it is actually `com.palantir.logsafe.Safe` and not just another class named `Safe`. Unless we are making a plain `Safe` marker a first-class concept in conjure in which case we should update the spec.

Reply:
See https://github.com/palantir/conjure-java/pull/255#discussion_r260938457

I don't feel strongly about it but I think the case where it is marked as safe but with another imported conjure class and that the expected behavior is to be different is quite convoluted. There is also some thoughts to make markers plain strings so this would be more aligned with it.

Reply:
also see https://github.com/palantir/conjure-java/pull/255#issuecomment-468175097 

btw, i'm refactoring the whole PR.

Reply:
I feel pretty strongly that for now we should mimic the current behavior of the Jersey resources, which look [specifically and only for `com.palantir.logsafe.Safe`](https://github.palantir.build/foundry/sls-logging/blob/5.2.0/sls-logging-jersey/src/main/java/com/palantir/sls/logging/RequestLoggingFilter.java#L199). I think changing this makes sense, but we should have a discussion about updating the spec before making decision like this in arbitrary PRs.

Reply:
@pkoenig10 That's the plan, this PR has been refactored so that we can retain the strictness of that check without special casing in conjure-undertow. wc will provide a `ParamMarker` implementation for `Safe.class`.